### ISSUE TITLE
Feature Request - per-user configuration on the Windower version

### ISSUE THREAD

**Issue by palladin9479** [Shirk/XIPivot#17]
Currently the widower version only loads data from a single global configuration XML stanza, even adding a new user entry just has it deleted the next time XIPivot loads.  Would be it be possible to have per-character definitions?  The use case is loading the nice HD graphic packs into only one player instance while allowing only the smaller paks to be loaded into mules and alts instances.

**Comment by palladin9479** [comment]
After seeing if I could edit it myself, it seems the offending line is
config.save(settings, 'all')

executed immediately after
settings = config.load(defaults)

Causing it to wipe out anything that isn't in the <global> section.  By commenting out the config.save it will now properly allow for per-character configurations.

**Comment by HealsCodes** [comment]
Your solution breaks settings in general if the user never had a pivot configuration though.

It feels like `config.save(settings, 'all')` replacing per-user config is a bug in the Windower config module..
Going by the [official documentation](https://github.com/Windower/Lua/wiki/Writing-Addons#addon-setup) settings should be saved globally at least once before they are available for per-user config:

> To save settings to file, there are two modes, a global and a per-character local mode. To save settings for one character only, call settings:save(), where settings contains the current settings structure. To save it globally for all characters, use settings:save('all').
> 
> **Note: a setting must be saved globally at least once before it can be saved for individual characters.**

I'm going to look into this though - thanks for bringing it to my attention.

**Comment by palladin9479** [comment]
Yeah it wasn't so much a solution but a confirmation of root cause.  I think a more elegant solution would be to check if the settings exist and if so read them, otherwise assume defaults and write out a default config.  Not sure if the windower addon API has a built in way to check for that file or if it would have to be built.

**Comment by State-Null** [comment]
Is this still an issue?
I tried setting rules for individual characters and they did not work
It only responded to global

**Comment by HealsCodes** [comment]
> Is this still an issue? I tried setting rules for individual characters and they did not work It only responded to global

Given the original issue looked like a problem with the Windower API I can't confirm or deny if it still is an issue.
I didn't add in any changes related to this so far.

**Comment by palladin9479** [comment]
Yes it's still an issue, config.save(settings, 'all') will always clobber the existing configuration because that is what it's supposed to do.  It writes the current values, which are defaults at the start, to the file.

I just commented it out and instead put

settings = config.load(defaults)

Near the top.  Then further down under 

windower.register_event('addon command', function(command, ...)

Look for 

config.save(settings)

And comment / replace it with

settings:save()

Those two changes tells it to load the defaults from the existing file., then later save your character context when you save. This will throw an error if there is no existing file, which is why most windower addons come with a settings.xml with generic values.  Since you already have a settings.xml, just do what I did above and you can then save your stuff per-character.
